pkg/util/shard.go

```diff
@@ -0,0 +1,17 @@
+package util
+
+import (
+	"crypto/md5"
+	"encoding/binary"
+)
+
+// ShuffleShardSeed returns seed for random number generator, computed from provided identifier.
+func ShuffleShardSeed(identifier string) int64 {
```

Comment:
Why not refactor and reuse? https://github.com/cortexproject/cortex/blob/master/pkg/querier/frontend/frontend_querier_queues.go#L226